static-instrumenter/agent-instrumenter/src/main/java/io/opentelemetry/contrib/statical/instrumenter/Main.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.statical.instrumenter;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.lang.instrument.ClassFileTransformer;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.jar.JarFile;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Main {
+
+  private static final String PATH_SEPARATROR = System.getProperty("path.separator");
+  private static final Logger logger = LoggerFactory.getLogger(Main.class);
+
+  private static final Main INSTANCE = new Main(new ClassArchive.Factory() {});
```

Comment:
Don't think we need this singleton, pass in dependencies to constructors appropriately

Reply:
I think we need it - the other part of the instrumentor runs in the java agent space and needs to store processed bytecode in the Main. So it's either full static or singleton and I opted for the latter. Makes sense for you? :)

Reply:
Hmm - I can imagine the opposite should also be possible, for the javaagent space to store the bytecode and Main reads them out when writing. Perhaps we can fix that when adding th elogic though.

Reply:
Agreed :)